cache/cache.go

```diff
@@ -0,0 +1,100 @@
+package cache
+
+import (
+	"container/list"
+	"main/repo"
+	"sync"
+)
+
+type CacheInterface interface {
+	Set(value *repo.Record)
+	Get(key string) *repo.Record
+	Delete(key string)
+}
+
+type InMemoCache struct {
+	cache map[string]*repo.Record
+	mu    sync.Mutex
+}
+
+func NewInMemoCache() *InMemoCache {
+	return &InMemoCache{cache: make(map[string]*repo.Record)}
+}
+
+func (c *InMemoCache) Set(value *repo.Record) {
+	c.mu.Lock()
+	c.cache[value.ID] = value
+	c.mu.Unlock()
+}
+
+func (c *InMemoCache) Get(key string) *repo.Record {
+	c.mu.Lock()
+	res, ok := c.cache[key]
+	c.mu.Unlock()
+	if !ok {
+		return nil
+	}
+	return res
+}
+
+func (c *InMemoCache) Delete(key string) {
+	c.mu.Lock()
+	delete(c.cache, key)
+	c.mu.Unlock()
+}
+
+type LruCache struct {
+	capacity int
+	queue    *list.List
+	cache    map[string]*Item
+}
+
+type Item struct {
+	data *repo.Record
+	key  *list.Element
+}
+
+func NewLruCache(capacity int) *LruCache {
+	return &LruCache{
+		capacity: capacity,
+		queue:    list.New(),
+		cache:    make(map[string]*Item),
+	}
+}
+
+func (c *LruCache) Set(value *repo.Record) {
+	item, ok := c.cache[value.ID]
+	if !ok {
```

Comment:
It's better to use direct `if` flow: 
instead of 
```go
if !ok {
    // do 1
} else {
    // do 2
}
```

use 
```go
if ok {
    // do 2
} else {
    // do 1
}
```

and early return can be used to remove else branch

```go
if ok  {
		item.data = value
		c.cache[value.ID] = item
		c.queue.MoveToFront(item.key)
                return
}
```